modules/top-level/plugins/default.nix

```diff
@@ -0,0 +1,56 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  inherit (import ./utils.nix lib)
+    normalizedPluginType
+    normalizePlugins
+    ;
+  byteCompileCfg = config.performance.byteCompileLua;
+in
+{
+  options = {
+    build.plugins = lib.mkOption {
+      visible = false;
+      internal = true;
+      readOnly = true;
+      type = lib.types.listOf normalizedPluginType;
```

Comment:
We _could_ consider moving the byte-compile impl here, using `apply`:

```nix
apply =
  if byteCompileCfg.enable then
    byteCompilePlugins
  else
    lib.id;
```

Reply:
Actually, I guess that doesn't work because we want to byte-comp _before_ combining?

Reply:
Yeah, if it's a read-only option then there's _not much_ difference between doing stuff in the definition vs the apply-fn.